dev-util/qbs/qbs-1.11.0.ebuild

```diff
@@ -0,0 +1,123 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit qmake-utils
+
+MY_P=${PN}-src-${PV}
+
+DESCRIPTION="Modern build tool for software projects"
+HOMEPAGE="https://wiki.qt.io/Qbs"
+SRC_URI="http://download.qt.io/official_releases/${PN}/${PV}/${MY_P}.tar.gz"
+
+LICENSE="|| ( LGPL-2.1 LGPL-3 )"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE="doc examples test"
+
+# see bug 581874 for the qttest dep in RDEPEND
+RDEPEND="
+	dev-qt/qtcore:5=
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtscript:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtxml:5
+	test? ( dev-qt/qttest:5 )
+"
+DEPEND="${RDEPEND}
+	doc? (
+		dev-qt/qdoc:5
+		dev-qt/qthelp:5
+	)
+	test? ( dev-qt/qtdeclarative:5 )
+"
+
+S=${WORKDIR}/${MY_P}
+
+src_prepare() {
+	default
+
+	# don't add /usr/include to INCLUDEPATH
+	# avoids a build failure in qt-creator with gcc-6 (bug 618424)
+	sed -i -e '/^INCLUDEPATH/ s:$${PWD}/\.\.::' src/lib/corelib/use_installed_corelib.pri || die
+
+	if ! use examples; then
+		sed -i -e '/INSTALLS +=/ s:examples::' static.pro || die
+	fi
+
+	if use test; then
+		sed -i -e '/SUBDIRS =/ s:=.*:= auto:' tests/tests.pro || die
+	else
+		sed -i -e '/SUBDIRS =/ d' tests/tests.pro || die
+	fi
+
+	# since 1.10, TestApi is either broken or requires more configuration
+	sed -i -e '/\<api\>/ d' tests/auto/auto.pro || die
+
+	# skip several tests that fail and/or have additional deps
+	sed -i \
+		-e 's/findArchiver("7z")/""/'		`# requires p7zip, fails` \
+		-e 's/findArchiver(binaryName,.*/"";/'	`# requires zip and jar` \
+		-e 's/p\.value("nodejs\./true||&/'	`# requires nodejs, bug 527652` \
+		-e 's/\(p\.value\|m_qbsStderr\.contains\)("typescript\./true||&/' `# requires nodejs and typescript` \
+		tests/auto/blackbox/tst_blackbox.cpp || die
+
+	# requires jdk, fails, bug 585398
+	sed -i -e '/blackbox-java\.pro/ d' tests/auto/auto.pro || die
+
+	# qbsres target uses the qbs binary to bootstrap itself so, with rpath disabled, we need to preload libqbscore.so
+	sed -i -e '\;qbsres.commands = ;a \    LD_PRELOAD=$$shell_quote($$shell_path($${QBS_LIBRARY_DIRNAME}/libqbscore.so)) \\' static-res.pro || die
```

Comment:
why not `LD_LIBRARY_PATH`?

Reply:
LD_PRELOAD doesn't trample on /etc/ld.so.conf and friends. So we get just the library we want without random breakages I've encountered when people go around changing LD_LIBRARY_PATH.
Qt Creator does this and it annoys me to no end to have to go into the project run settings every time to unset LD_LIBRARY_PATH which always gets nuked to '/usr/lib64:/path/to/my/build/directory' causing all sorts of fun breakages like the wrong GL library being selected.

Reply:
I mainly prefer LD_PRELOAD because I find it's easier for me to get it right. With LD_PRELOAD there is minimal interfering with the environment and therefore fewer unknown variables.

If ebuild is already setting an LD_LIBRARY_PATH, or if there is some tool to dump the current effective LD_LIBRARY_PATH, and it's just a question of inserting or appending to that I have no preference either way.

Reply:
`LD_LIBRARY_PATH` is already being used by this ebuild.